bitcoin/src/blockdata/witness.rs

```diff
@@ -68,20 +68,26 @@ fn fmt_debug(w: &Witness, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error>
     f.write_str("witnesses: [")?;
 
     let instructions = w.iter();
-    let last_instruction = instructions.len() - 1;
-
-    for (i, instruction) in instructions.enumerate() {
-        let bytes = instruction.iter();
-        let last_byte = bytes.len() - 1;
-
-        f.write_str("[")?;
-
-        for (j, byte) in bytes.enumerate() {
-            write!(f, "{:#04x}", byte)?;
-            f.write_str(comma_or_close(j, last_byte))?;
+    if instructions.len() > 0 {
+        let last_instruction = instructions.len() - 1;
```

Comment:
I think

```
match instructions.len().checked_sub(1) {
  Some(last_instruction) => ...
  None => ..
}
```

would be more refactor-solid and better convey the underflow risk

Reply:
Good suggestion, I'll do up another PR and let this one merge. Thanks!

Reply:
Done in #2035